asynction/server.py

```diff
@@ -176,35 +178,68 @@ def from_spec(
             if url_parse_result.path:
                 kwargs["path"] = url_parse_result.path
 
+            server_security = server.security
+
         asio = cls(spec, validation, docs, app, **kwargs)
-        asio._register_handlers(default_error_handler)
+        asio._register_handlers(default_error_handler, server_security=server_security)
         return asio
 
     def _register_namespace_handlers(
         self,
         namespace: str,
-        channel_handlers: ChannelHandlers,
+        channel_handlers: Optional[ChannelHandlers],
         channel_bindings: Optional[ChannelBindings],
+        server_security: Optional[Sequence[SecurityRequirement]] = None,
     ) -> None:
-        if channel_handlers.connect is not None:
-            handler = load_handler(channel_handlers.connect)
+
+        on_connect = None
+
+        # if a connection handler is defined then load it
+        if channel_handlers and channel_handlers.connect is not None:
+            on_connect = load_handler(channel_handlers.connect)
 
             if self.validation:
                 with_bindings_validation = bindings_validator_factory(channel_bindings)
-                handler = with_bindings_validation(handler)
+                on_connect = with_bindings_validation(on_connect)
+
+        if server_security:
+            # if a connection handler was not defined and sever_security is present
+            # then we will inject a default handler to apply security
+            if not on_connect:
 
-            self.on_event("connect", handler, namespace)
+                def _on_connect(*args, **kwargs):
```

Comment:
Not very keen on this implicit variable assignment.

Let's move the [`_noop_handler`](https://github.com/dedoussis/asynction/blob/main/asynction/mock_server.py#L115) defined in the `mock_server.py` module within `server.py`, and have `mock_server` importing it from `server`.

Then we can just initialise the `on_connect` with `_noop_handler`, and override it if `channel_handlers.connect` is defined.